doc/userguide/performance/high-performance-config.rst

```diff
@@ -1,19 +1,341 @@
+ /usr/local/sbin/ethtool -G eth1 rx 1024
+
+Make sure the RSS hash function is Toeplitz:
+
+::
+
+ /usr/local/sbin/ethtool -X eth1 hfunc toeplitz
+ 
+Let the NIC balance as much as possible:
+
+::
+
+ for proto in tcp4 udp4 tcp6 udp6; do
+    /usr/local/sbin/ethtool -N eth1 rx-flow-hash $proto sdfn
+ done
+
+and in the af-packet section of suricata.yaml: 
+
+::
+
+ af-packet:
+  - interface: eth1
+    # Number of receive threads. "auto" uses the number of cores
+    threads: 16
+    # Default clusterid. AF_PACKET will load balance packets based on flow.
+    cluster-id: 99
+    # Default AF_PACKET cluster type. AF_PACKET can load balance per flow or per hash.
+    # This is only supported for Linux kernel > 3.1
+    # possible value are:
+    #  * cluster_flow: all packets of a given flow are send to the same socket
+    #  * cluster_cpu: all packets treated in kernel by a CPU are send to the same socket
+    #  * cluster_qm: all packets linked by network card to a RSS queue are sent to the same
+    #  socket. Requires at least Linux 3.14.
+    #  * cluster_ebpf: eBPF file load balancing. See doc/userguide/capture-hardware/ebpf-xdp.rst for
+    #  more info.
+    # Recommended modes are cluster_flow on most boxes and cluster_cpu or cluster_qm on system
+    # with capture card using RSS (require cpu affinity tuning and system irq tuning)
+    cluster-type: cluster_qm
+    ...
+    ...
+
+CPU affinity and NUMA
+---------------------
+
+Intel based systems
+~~~~~~~~~~~~~~~~~~~
+
+If the system has more then one NUMA node there are some more possibilities. In those cases it is generally recommended to use as many worker threads as cpu cores available/possible - from the same NUMA node. The example below uses 72 core machine and the sniffing NIC that Suricata uses located on NUMA node 1.
+
+In the case below we are using 36 worker threads out of NUMA node 1's CPU, af-packet runmode with `cluster-type: cluster_qm`.
+
+If the CPU's NUMA set up is as follows:
+
+::
+
+    lscpu
+    Architecture:        x86_64
+    CPU op-mode(s):      32-bit, 64-bit
+    Byte Order:          Little Endian
+    CPU(s):              72
+    On-line CPU(s) list: 0-71
+    Thread(s) per core:  2
+    Core(s) per socket:  18
+    Socket(s):           2
+    NUMA node(s):        2
+    Vendor ID:           GenuineIntel
+    CPU family:          6
+    Model:               79
+    Model name:          Intel(R) Xeon(R) CPU E5-2697 v4 @ 2.30GHz
+    Stepping:            1
+    CPU MHz:             1199.724
+    CPU max MHz:         3600.0000
+    CPU min MHz:         1200.0000
+    BogoMIPS:            4589.92
+    Virtualization:      VT-x
+    L1d cache:           32K
+    L1i cache:           32K
+    L2 cache:            256K
+    L3 cache:            46080K
+    NUMA node0 CPU(s):   0-17,36-53
+    NUMA node1 CPU(s):   18-35,54-71
+
+It is recommended that 36 worker threads are used and the NIC set up could be as follows:
```

Comment:
There is some discussion to keep at least the first core of a CPU free for some "management" stuff. Especially if there is not just the NIC on that node, for example also the Disks and/or the NICs used for ssh connection.